i_camera_in_manuell

schmitzdominic opened the pull request:
- Es ist nun möglich ein Foto zu machen und dieses wird dann in die manuelle Maske übertragen
- Recognition Activity wurde gelöscht
- Bilder können nun ausgewählt werden und werden über ocr ausgewertet
- Default matcher hat nun für die Methoden ArrayList rückgaben

schmitzdominic committed:
Alle features für i_camera_in_manuell implementiert

- Es ist nun möglich ein Foto zu machen und dieses wird dann in die manuelle Maske übertragen
- Recognition Activity wurde gelöscht
- Bilder können nun ausgewählt werden und werden über ocr ausgewertet
- Default matcher hat nun für die Methoden ArrayList rückgaben

schmitzdominic committed:
Code effizienter gestaltet.

joeymez commented on app/src/main/java/de/projektss17/bonpix/A_Main.java:
```diff
@@ -140,7 +140,7 @@ public void onClick(View view) {
             @Override
             public void onClick(View view) {
-                S.showManuell(A_Main.this);
+                S.showManuell(A_Main.this,"new");
```
Was genau soll der String "New" derzeit bewirken? es wird im Prinzip ein String mit übergeben beim öffnen der neuen Activity. Dennoch verstehe ich noch nicht ganz wieseo "New"?

schmitzdominic replied:
Hi Johanns :) 
Hab der Methode showManuell einen weiteren Parameter Status übergeben damit bei der erstellung der Activity immer klar ist welchen Status die Maske besitzen soll. "new" bedeutet eine leere neue Maske, "edit" bedeutet das ein Bon editiert wird und "foto" bedeutet das die Maske nach Aufnahme eines Fotos geöffnet wird (pfad wird übergeben und foto wird angezeigt bzw. gleich ausgelesen)

joeymez commented on app/src/main/java/de/projektss17/bonpix/A_Main.java:
```diff
@@ -422,7 +422,7 @@ protected void onActivityResult(int requestCode, int resultCode, Intent data) {
             String picturePath = cursor.getString(column_index_data);
             picturePathList.add(picturePath);
-            S.showRecognition(A_Main.this,picturePathList);
+            S.showManuell(A_Main.this, picturePathList, "foto");
```
Auch hier ist es mir noch nicht ganz bewusst was genau der string "Foto" bringen soll. Sind diese Stings lediglich Beschreibung für irgendetwas? Indiesem Fall für den ArrayString? :)

schmitzdominic replied:
Siehe vorheriges Kommentar

joeymez commented on app/src/main/java/de/projektss17/bonpix/A_OCR_Manuell.java:
```diff
@@ -669,4 +695,111 @@ public boolean checkAllRelevantValues(){
+
+        for(int i = 0; i < linearLayout.getChildCount(); i++){
+            view = linearLayout.getChildAt(i);
+            if(view.findViewById(R.id.ocr_manuell_article_line) != null){
```
Ich habe den Branch sowohl auf meinem PC als auch auf meinem Laptop ausgecheckt und gefetcht. In beiden Fällen wir diese Zeile als Fehlerhaft angezeigt.....

>> Die ID "R.id.ocr_manuell_article_line" ist nicht vergeben. Ich schätze die ObjektID wurde umbennant. Bitte umändern bevor ich ein OK geben kann :)

schmitzdominic replied:
Die ID existiert, du musst nur dein gradle nocheinmal neu syncen ;) Ich muss allerdings gestehen das die Zeile bei mir auch ab und zu Probleme macht, total komisch, weil die id existiert.. Hatte früher immer das CammelCase in verdacht aber an dem liegts wohl nicht.

joeymez reviewed (changes requested):
Meiner Seitens wurde nur ein Fehler gefunden. Der Rest ist meiner Meinung nach in Ordnung :) Dennoch sollte der Fehler ausgebessert werden, welches ich gefunden habe.

Sobald ausgebessert gibt es ein OK von mir ;)

dontcallme reviewed (changes requested):
Hab die App mal auf funktion getestet e_camera_ocr (S3 mini Android 4.1.2)

Folgende Probleme sind mir aufgefallen:
- Wenn Auto-Drehen aktiv war beim bild machen, richtet die App das bild falsch aus --> OCR erkennt nichts richtig + Bild wird z.B in ocr manuell auch verkehrt herum angezeigt
- Manchmal stürtzt die App in OCR-Manuell beim Bild auswählen ab (kein Muster erkennbar)
- Bei Manueller auswahl des Bilds wird bei mir nichts ausgewertet 
- Wenn das Foto mit der App gemacht wird erfolgt bei mir die ausgabe noch in dem Ausgabe Fenster
- !!Der Speicher Button von OCR_Manuell wird nicht angezeigt (evtl. ein Auflösungsproblem)

Hatte ldr. noch nicht mehr Zeit, um die Fehler im Code zu suchen

schmitzdominic commented:
Hi Domi, 
Danke das du dir die Zeit genommen hast ;)

- Das mit dem Drehen der Kamera müssen wir generell Mal deaktivieren, das hat aber momentan nichts mit diesem Item zu tun

- Ich kann leider kein anderes Problem nachstellen bzw. reproduzieren, daher kann ich es schlecht beheben. Das du noch die alte Activity nach dem Foto machen angezeigt bekommst, könnte ein Indiz dafür sein das du eventuell nicht den neuesten Stand hast? Bitte fetch nochmal und probiers mal mit nem anderen Handy. Danke :)

dontcallme reviewed (changes requested):
SO bei mir hats jetzt auch geklapp :)
Übertragung der Daten funktioniert soweit (Manuell + Foto)
- Anzumerken wäre nur noch, das der Laden nicht automatisch mit ausgewählt wird (bitte kurze Rückmeldung, liegt denk noch ander fehlenden DB)
- Auch würde ich die Zählweise bei den Artikel bei 1 starten lassen (Artikel 0 geht zwar, find ich aber irgendwie seltsam)
- Habs leider im Code nicht gefunden --> würde den Fokus, wenn die Daten über das Foto/Bild eingefügt werden rausnehmen, sonst landet man immer im letzten feld.

schmitzdominic commented:
- Die Anmerkung ist gerechtfertigt, allerdings beheben wir das in dem item i_datenbank_verbessern in diesem sprint ;)

- Das mit der Zählweise bekomm ich hin, bin leider nur nicht am Rechner momentan, ich geb dir bescheid wenn ichs push :) WURDE GEÄNDERT: https://github.com/schmitzdominic/ProjektSS17/pull/169/commits/1a7687977fe7c951f6f83cc8e7eb10e7523aa376

- Das mit dem Fokus könnte schwer werden da der Fokus automatisch beim erstellen eines neuen Items hin springt xD Daher würde ich es eventuell auch in einem anderen Sprint später behandeln. WURDE GEÄNDERT: https://github.com/schmitzdominic/ProjektSS17/pull/169/commits/1a7687977fe7c951f6f83cc8e7eb10e7523aa376

schmitzdominic committed:
Gemäß Review änderungen vorgenommen

- Focus springt nun nicht mehr nach dem hinzufügen eines bildes zum letzten Produkt
- Artikel fangen nun bei 1 an zu zählen

dontcallme commented on app/src/main/java/de/projektss17/bonpix/A_OCR_Manuell.java:
```diff
@@ -776,7 +784,7 @@ public void doState(String state){
+            articleArray[count] = new C_Article("Article "+(count+1), price);
             count++;
         }
 
```
Top, jetzt startets auch mit Artikel 1 :)

dontcallme reviewed (approved):
Gute Arbeit Domi :) , 
- fängt jetzt bei 1 an zu zählen
- Fokus springt nicht mehr automatisch rein

Von mir gibts ein OK, hab sonst nichts mehr gefunden.
